pkg/es/client/index_client.go

```diff
@@ -0,0 +1,255 @@
+// Copyright (c) 2021 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package client
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+)
+
+// Index represents ES index.
+type Index struct {
+	// Index name.
+	Index string
+	// Index creation time.
+	CreationTime time.Time
+	// Aliases
+	Aliases map[string]bool
+}
+
+// Alias represents ES alias.
+type Alias struct {
+	// Index name.
+	Index string
+	// Alias name.
+	Name string
+	// IsWritedIndex option
+	IsWriteIndex bool
+}
+
+// IndicesClient is a client used to manipulate indices.
+type IndicesClient struct {
```

Comment:
Could you please mark what this struct implements? 


```go
var _ IndexAPI = (*IndicesClient)(nil)
```

applies also for other structs in this PR.